libraries/chain/webassembly/crypto.cpp

```diff
@@ -246,4 +249,88 @@ namespace eosio { namespace chain { namespace webassembly {
       return return_code::success;
    }
 
+   bool interface::bls_verify( span<const char> signature, span<const char> digest, span<const char> pub) const {
```

Comment:
From what I can tell `digest` here is a misnomer: it's actually a message.

It's not clear to me if there would be utility in a separate host function for mapping the message to the group, so that this host function would already take only a mapped point.

As a message, a couple things:

There is no need to stuff the message in a `vector` and then unpack that, the `span` type already defines the length of the message because the interface deconstructs to a `char* message, size_t message_size` effectively.

Any host function that processes an arbitrary length of bytes needs to be mindful about taking too long because host functions aren't interruptible. You'll see in, for example, the `sha256` host function how it periodically checks to see if it should abort due to time constraints. Since this function is hashing an arbitrary length message, you will need to do one of:
* periodically check for time. This is ideal, but implies changes down to Relic I assume. Or,
* Add some sort of subjective check to the size of the message (see the `mod_exp` host function as a good recent example)